src/components/Select/Select.jsx

```diff
@@ -119,18 +133,20 @@ export default class Select extends React.Component {
           })
         : items;
 
-    const optionIndex = deselectItems.findIndex(x => x.id === option.id);
+    const optionIndex = deselectItems.findIndex(
+      x => this.getItemId(x) === this.getItemId(option)
+    );
 
     this.setState({
       isOpen: multiSelect,
-      items:
-        optionIndex > -1
-          ? helper(deselectItems, {
-              [optionIndex]: {
-                selected: { $set: !option.selected },
-              },
-            })
-          : items,
+      items: this.updateItemsInState(deselectItems, option, optionIndex),
+      // we need to copy the items to another array here in order to use that one during search
+      // filtering
+      selectedItems: this.updateItemsInState(
+        deselectItems,
```

Comment:
whenever we're selecting/deselecting an item, update the `selectedItems` as well in order to be able to use it later when searching/filtering